File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) (res int) {
	sum := n * (n + 1) / 2
	return sum * sum
}

func SumOfSquares(n int) (res int) {
	return n * (n + 1) * (2*n + 1) / 6
}

func Difference(n int) int {
	nn := n * n
	return (3*nn + n + n) * (nn - 1) / 12
}

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(value)
    @numbers = (1..value)
  end

  def square_of_sum
    @numbers.sum.abs2
  end

  def sum_of_squares
    @numbers.map(&:abs2).sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
